Automattic/wp-calypso
# simplify DSP request handlers and create a request handler for the /campaigns/:id route

Related to #

## Proposed Changes

- Simplified and centralized the handler to call requests to the DSP so it could be reused
- Updated the previous handler in /campaigns route to the new system
- Created a new handler in /campaigns/:id route to create a DSP user if the DSP returns "no_local_user" error in DSP

## Testing Instructions
- Bridge the dsp instance to calypso (changing wordads-dsp-proxy, routing the SSH and running the SSH tunnel)
- Go to any campaign details page (/campaigns/:id route)
- you should see the data. Now remove your dsp_user and dsp_social_logins user in your DSP local database and refresh
- After a few seconds you should see the campaign data. A new DSP user should've created while you tried to fetch the campaign data
- Try the same in /campaigns page. It should display the same behaviour


## Pre-merge Checklist
